jms/src/main/scala/akka/stream/alpakka/jms/JmsMessages.scala

```diff
@@ -70,22 +70,27 @@ object JmsPassThrough {
 }
 
 /**
- * Marker trait for stream elements that do not contain pass-through data.
+ * Marker trait for stream elements that contain pass-through data with properties.
  */
-sealed trait JmsMessage extends JmsEnvelope[NotUsed] {
-
-  def withHeader(jmsHeader: JmsHeader): JmsMessage
+sealed trait JmsEnvelopeWithProperties[+PassThrough] extends JmsEnvelope[PassThrough] {
```

Comment:
I _think_ it's required because it's used in pass through JMS messages. If I omit it then type param errors with the `JmsMessage` pass through factory methods.